### PR TITLE
Fix4924 discrete heatmap2d

### PR DESCRIPTION
**Pull request by ordiology**
This version of gl-heatmap2d modifies Mikola Lysenko's original code to optionally produced a discretised heatmap instead of an interpolated one. This is achieved by introducing an is.smooth variable:

 var isSmooth = options.zsmooth !== false

Scientific data is often discretised and this option will allow the data to be represented as measured rather than smoothing between observations.

This change will resolve this plotly issue: https://github.com/plotly/plotly.js/issues/4924, and is currently being addressed in this pull: https://github.com/plotly/plotly.js/pull/4953

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,3 +16,5 @@ node_modules/*
 package-lock.json
 *.DS_Store
 gh-pages
+.Rproj.user
+gl-heatmap2d.Rproj
```
Comment (archmoj) [inline .gitignore]: Why we need to add these?

**Comment by archmoj** [comment]
@ordiology thanks very much for the PR.
Would you mind adding info to the README file regarding `zsmooth: false` option.
This is going to be released in a minor.

File: .gitignore [hunk 2]
```diff
@@ -9,10 +9,4 @@ lib-cov
 
 pids
 logs
-results
-
-npm-debug.log
-node_modules/*
-package-lock.json
```
Comment (archmoj) [inline .gitignore:16]: Let's revert these changes in `.gitignore` file.

File: README.md [hunk 3]
```diff
@@ -2,5 +2,16 @@ gl-heatmap2d
 ============
 2D heatmaps
 
+Modified by: Louise Ord
```
Comment (archmoj) [inline README.md]: How about mentioning this info in a sentence right after the following line?
`This is achieved by introducing an option, zsmooth, that defaults to the smoothed heatmap.`

File: .gitignore (hunk 2 above)
Comment (ordiology) [reply .gitignore:16]: I've tried to do this, but I'm not sure I've done it correctly. Please let me know if there's still an issue.

**Comment by archmoj** [comment]
@ordiology thanks very much for th PR.
:dancer:
